### PR TITLE
Override _metatize in TFlowMetaOp to fix inputs

### PR DESCRIPTION
**Pull request by josephwillard**
Using `_reconstruct_sequence_inputs` we can make sure that inputs no longer show up as a flattened list. This is related to #28.

### CODE REVIEW

File: symbolic_pymc/tensorflow/meta.py [hunk 1]
```diff
@@ -337,6 +337,13 @@ def process_node_def(cls, node_def, op_def, obj=None):
 
         return meta_node_def
 
+    @classmethod
+    def _metatize(cls, obj):
+        """Fix inputs from TF"""
+        obj.input = obj._reconstruct_sequence_inputs(obj.op_def,
```
Comment (brandonwillard) [inline symbolic_pymc/tensorflow/meta.py]: We should not have to alter TF objects for most&mdash;if not all&mdash;of the functionality in this project, so this assignment will require a lot of justification.

More importantly, this overloaded method isn't implemented correctly and will break `metatize` functionality for `tf.Operation` objects.  It needs to return a meta object.

File: symbolic_pymc/tensorflow/meta.py [hunk 2]
```diff
@@ -340,10 +340,13 @@ def process_node_def(cls, node_def, op_def, obj=None):
...
         """Fix inputs from TF"""
-        obj.input = obj._reconstruct_sequence_inputs(obj.op_def,
-                                                      obj.inputs,
-                                                      obj.node_def.attr)
-    
+        new_input = obj._reconstruct_sequence_inputs(obj.op_def,
+                                                     obj.inputs,
+                                                     obj.node_def.attr)
+        new_args = [getattr(obj, s) for s in getattr(cls, "__slots__", [])]
+        new_args[-2] = new_input
```
Comment (brandonwillard) [inline symbolic_pymc/tensorflow/meta.py]: This shouldn't have to rely on the order of elements in `__slots__`, nor use a value that is immediately replaced.  Instead, you could check for when `s == "inputs"` in the list comprehension and use `new_input` in the first place.

File: symbolic_pymc/tensorflow/meta.py [hunk 3]
```diff
@@ -337,6 +337,13 @@ def process_node_def(cls, node_def, op_def, obj=None):
 
         return meta_node_def
 
+    @classmethod
+    def _metatize(cls, obj):
+        """Fix inputs from TF"""
```
[the remark this answers is not in the document]
Comment (josephwillard) [reply symbolic_pymc/tensorflow/meta.py]: Fixed in latest commit.

File: symbolic_pymc/tensorflow/meta.py (hunk 2 above)
Comment (josephwillard) [reply symbolic_pymc/tensorflow/meta.py]: Fixed in latest commit.

File: tests/utils.py [hunk 4]
```diff
@@ -2,22 +2,28 @@
 
 
 def assert_ops_equal(a, b, compare_fn=lambda a, b: a.op.type == b.op.type):
+    # TODO: Add a test that determines if a is a meta object
```
Comment (brandonwillard) [inline tests/utils.py]: This TODO needs to be resolved.

File: tests/utils.py [hunk 5]
```diff
@@ -2,22 +2,28 @@
...
+        assert len(a.op.inputs) == len(new_inputs)
 
         if isinstance(a.op, Mapping):
             a_inputs = list(a.op.inputs.values())
         else:
             a_inputs = list(a.op.inputs)
 
         if isinstance(b.op, Mapping):
-            b_inputs = list(b.op.inputs.values())
+            b_inputs = new_inputs#list(b.op.inputs.values())
```
Comment (brandonwillard) [inline tests/utils.py]: This comment needs to be removed.

File: tests/utils.py [hunk 6]
```diff
@@ -2,22 +2,28 @@
...
         else:
-            b_inputs = list(b.op.inputs)
+            b_inputs = new_inputs#list(b.op.inputs)
```
Comment (brandonwillard) [inline tests/utils.py]: This comment needs to be removed.

File: tests/utils.py [hunk 7]
```diff
@@ -2,22 +2,28 @@
...
 def assert_ops_equal(a, b, compare_fn=lambda a, b: a.op.type == b.op.type):
+    # TODO: Add a test that determines if a is a meta object
     if hasattr(a, 'op') or hasattr(b, 'op'):
         assert hasattr(a, 'op') and hasattr(b, 'op')
 
         assert compare_fn(a, b)
 
-        assert len(a.op.inputs) == len(b.op.inputs)
+        # grabing updated input representation
+        new_inputs = b.op._reconstruct_sequence_inputs(b.op.op_def,
```
Comment (brandonwillard) [inline tests/utils.py]: As the TODO above states, you need to check whether or not the inputs even need to be restructured (i.e. check if `a` and/or `b` is a TF tensor and *then* restructure accordingly).

**Pull request merged**